### PR TITLE
AccountManager::update_pre_key_bundle

### PR DESCRIPTION
**Pull request by rubdos**
I was looking at https://github.com/gferon/signal-bot-rs/blob/wip/src/manager.rs#L361 and figured this should be common code (as it is in the Java code).

I have integrated this in Whisperfish, but haven't tested yet. Will do that tomorrow.

Good night! ~R

### CODE REVIEW

File: libsignal-service/src/account_manager.rs [hunk 1]
```diff
@@ -1,15 +1,25 @@
...
 }
 
+const PRE_KEY_MINIMUM: u32 = 10;
+const PRE_KEY_BATCH_SIZE: u32 = 100;
+
 impl<Service: PushService> AccountManager<Service> {
-    pub fn new(service: Service) -> Self {
-        Self { service }
+    pub fn new(context: Context, service: Service) -> Self {
+        Self { service, context }
```
Comment (gferon) [inline libsignal-service/src/account_manager.rs:22]: being pedantic here, but maybe swap the order of the fields to match the constructor?

**Review by gferon — APPROVED** [review]
Awesome! I only have one minor nitpick. It's really nice to integrate more common code now that we start having different consumers of the crate.

**Review by gferon — APPROVED** [review]
Just fixed the build of the example, will merge when CI is green :rocket:

File: libsignal-service/src/account_manager.rs [hunk 2]
```diff
@@ -42,4 +52,57 @@ impl<Service: PushService> AccountManager<Service> {
...
+    /// Checks the availability of pre-keys, and updates them as necessary.
+    ///
+    /// Parameters are the protocol's `StoreContext`, and the offsets for the next pre-key and
+    /// signed pre-keys.
+    ///
+    /// Equivalent to Java's RefreshPreKeysJob
+    pub async fn update_pre_key_bundle(
+        &mut self,
+        store_context: StoreContext,
+        pre_keys_offset_id: u32,
```
Comment (gferon) [inline libsignal-service/src/account_manager.rs:65]: What about outputting the new value (so the caller doesn't have to know how many pre-keys we generate, if we did that etc.) as either a return value, or taking the parameter as `&mut pre_keys_offset_id`?

Comment (rubdos) [reply libsignal-service/src/account_manager.rs:65]: Good idea. Whisperfish (pre 0.5) used a random u32 as next pre-key ID, and I just look at those that I have stored and go from there. But returning it is very useful indeed.

File: libsignal-service/src/account_manager.rs (hunk 1 above)
Comment (rubdos) [reply libsignal-service/src/account_manager.rs]: I was wondering whether the `StoreContext` from below should maybe also go in the constructor.

**Pull request merged**